platform/server/detect.py

```diff
@@ -100,6 +102,27 @@ def configure(env):
         env.extra_suffix = ".llvm" + env.extra_suffix
         env.Append(LIBS=["atomic"])
 
+    # Linker
+    if env["linker"] != "default":
```

Comment:
i think you need to separate this part to few functions that will handle each  case, currently its way to complicated for one functions, stick to SRP and try to make each function do only one thing, also think about OCP, if we want to append this part? how difficult it will be? if a module will change, do we need to rewrite the whole function or you can create a mini function that will do this part? also, use consts for constant strings so if it will be changed in the future you can just replace it in the beginning instead of rewriting that in every part it appears.